src/probnum/filtsmooth/statespace/sde.py

```diff
@@ -18,26 +20,37 @@ class SDE(transition.Transition):
     driven by a Wiener process with unit diffusion.
     """
 
-    def __init__(self, driftfun, dispmatrixfun, jacobfun):
-        self._driftfun = driftfun
-        self._dispmatrixfun = dispmatrixfun
-        self._jacobfun = jacobfun
-
-    def transition_realization(self, real, start, stop, **kwargs):
+    def __init__(
+        self,
+        driftfun: Callable[[FloatArgType, np.ndarray], np.ndarray],
+        dispmatfun: Callable[[FloatArgType, np.ndarray], np.ndarray],
+        jacobfun: Callable[[FloatArgType, np.ndarray], np.ndarray],
+    ):
+        self.driftfun = driftfun
+        self.dispmatfun = dispmatfun
+        self.jacobfun = jacobfun
+        self.precon = None
+
+    def transition_realization(
+        self,
+        real,
+        start,
+        stop=None,
+        step=None,
+        linearise_at=None,
+    ):
         raise NotImplementedError
 
-    def transition_rv(self, rv, start, stop, **kwargs):
+    def transition_rv(
+        self,
+        rv,
+        start,
+        stop=None,
+        step=None,
+        linearise_at=None,
+    ):
```

Comment:
since those remain (abstract) interfaces, I use the full signature and no `kwargs`.